models/compat/wakatime/v1/leaders.go

```diff
@@ -0,0 +1,97 @@
+package v1
+
+import (
+	"github.com/duke-git/lancet/v2/slice"
+	"github.com/muety/wakapi/models"
+	"time"
+)
+
+// partially compatible with https://wakatime.com/developers#leaders
+
+type LeaderboardViewModel struct {
+	CurrentUser CurrentUser        `json:"current_user"`
+	Data        []LeaderboardEntry `json:"data"`
+}
+
+type CurrentUser struct {
+	Rank int   `json:"rank"`
+	Page int   `json:"page"`
+	User *User `json:"user"`
+}
+
+type LeaderboardEntry struct {
+	Rank         int          `json:"rank"`
+	RunningTotal RunningTotal `json:"running_total"`
+	User         *User        `json:"user"`
+}
+
+type RunningTotal struct {
+	TotalSeconds              float64    `json:"total_seconds"`
+	HumanReadableTotal        string     `json:"human_readable_total"`
+	DailyAverage              float64    `json:"daily_average"`
+	HumanReadableDailyAverage string     `json:"human_readable_daily_average"`
+	Languages                 []Language `json:"languages"`
+}
+
+type Language struct {
+	Name         string  `json:"name"`
+	TotalSeconds float64 `json:"total_seconds"`
+}
+
+func NewRunningTotal(entry *models.LeaderboardItemRanked) RunningTotal {
+	return RunningTotal{
+		TotalSeconds:              entry.Total.Seconds(),
+		HumanReadableTotal:        entry.Total.String(),
+		DailyAverage:              entry.Total.Seconds() / 7,
+		HumanReadableDailyAverage: (entry.Total / 7).String(),
+		Languages:                 []Language{},
+	}
+}
+
+func NewLeaderboardEntry(entry *models.LeaderboardItemRanked) LeaderboardEntry {
+	return LeaderboardEntry{
+		Rank:         int(entry.Rank),
+		RunningTotal: NewRunningTotal(entry),
+		User:         NewFromUser(entry.User),
+	}
+}
+
+func NewLeaderboardViewModel(leaderboard *models.Leaderboard, user *models.User) *LeaderboardViewModel {
+	var entries []LeaderboardEntry
+	var currentUserEntry = CurrentUser{}
+	for _, entry := range *leaderboard {
+		if user != nil && entry.User.ID == user.ID {
+			currentUserEntry = CurrentUser{
+				Rank: int(entry.Rank),
```

Comment:
Wouldn't this overwrite the user's rank in every iteration of the loop? Say your current user is overall at rank 1, for language "Go" at rank 4 and for language "Java" at rank 3. If the "Java" rank appears last in the list of entries, the user's rank would be falsely shown as 3, no? I might be confusing things here, but please briefly think through this again.

Also, instantiating a new object for every entry of that user seems a fit inefficient.